contracts/modules/cover/Cover.sol

```diff
@@ -260,33 +261,39 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
           / (allocationRequest.previousExpiration - allocationRequest.previousStart); // previous period
       }
 
-      // converting asset amount to nxm and rounding up to the nearest NXM_PER_ALLOCATION_UNIT
-      uint coverAmountInNXM = Math.roundUp(
-        Math.divCeil(poolAllocationRequests[i].coverAmountInAsset * ONE_NXM, nxmPriceInCoverAsset),
-        NXM_PER_ALLOCATION_UNIT
-      );
-
-      uint premiumInNXM = stakingPool(poolAllocationRequests[i].poolId).requestAllocation(
-        coverAmountInNXM,
-        vars.previousPremiumInNXM,
-        allocationRequest
-      );
-
-      // omit deallocated pools from the segment
-      if (coverAmountInNXM != 0) {
-        coverSegmentAllocations[allocationRequest.coverId][segmentId].push(
-          PoolAllocation(
-            poolAllocationRequests[i].poolId,
-            coverAmountInNXM.toUint96(),
-            premiumInNXM.toUint96()
-          )
+      {
```

Comment:
The added scope is redundant here as the block is the latest in the for loop block and it won't help to avoid a stack too deep error. Remove the added scope.